app/scripts/contentscript.js

```diff
@@ -156,16 +210,32 @@ const destroyPhishingExtStreams = () => {
 };
 
 /**
- * Resets the extension stream with new streams to channel with the phishing page streams,
- * and creates a new event listener to the reestablished extension port.
+ * This listener destroys the phishing extension streams when the extension port is disconnected,
+ * so that streams may be re-established later the phishing extension port is reconnected.
  */
-const resetPhishingStreamAndListeners = () => {
-  phishingExtPort.onDisconnect.removeListener(resetPhishingStreamAndListeners);
-
+const onDisconnectDestroyPhishingStreams = () => {
+  phishingExtPort.onDisconnect.removeListener(
+    onDisconnectDestroyPhishingStreams,
+  );
   destroyPhishingExtStreams();
-  setupPhishingExtStreams();
+};
 
-  phishingExtPort.onDisconnect.addListener(resetPhishingStreamAndListeners);
+/**
+ * When the extension background is loaded it sends the EXTENSION_MESSAGES.READY message to the browser tabs.
+ * This listener/callback receives the message to set up the streams after service worker in-activity.
+ *
+ * @param {object} msg
+ * @param {string} msg.name - custom property and name to identify the message received
+ * @returns {Promise|false}
+ */
+const onMessageSetUpPhishingStreams = (msg) => {
+  if (msg.name === EXTENSION_MESSAGES.READY) {
+    setupPhishingExtStreams();
+    return Promise.resolve(
```

Comment:
Is there an established style preference for this pattern?

cf https://github.com/MetaMask/metamask-extension/pull/16221

Reply:
Thanks for the link. I dig the proposed `async` > `Promise.resolve` pattern

The thing with the onMessage callback is that we either want to return a Promise, boolean, or undefined as mentioned in the [onMessage doc](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage), so that we can handle specific message types: 

> If you only want the listener to respond to messages of a certain type, you must define the listener as a non-async function, and return a Promise only for the messages the listener is meant to respond to — and otherwise return false or undefined:

Reply:
That makes perfect sense, thanks for laying that out. Would you be open to changing from `return false` or `return undefined` below? Seems more semantically accurate and would make the intention more obvious to future readers not aware of the nuance of boolean being a specific case.

Reply:
I like this reasoning! what if we save some lines and let the function return undefined by default? 

updated: [bb5ba6b](https://github.com/MetaMask/metamask-extension/pull/16075/commits/bb5ba6b2e1c4d066a96fc8d1f4885d6d1cb6f2b9) and [0f9ff27](https://github.com/MetaMask/metamask-extension/pull/16075/commits/0f9ff27c0510ab78d6fd3f76007ba05abdba0b4a)

Reply:
@legobeat 

I like this suggestion! thank you. updated the code to return `undefined` instead of `false`